reference/plugins.md

```diff
@@ -2,7 +2,168 @@
 layout: page
 title: Plugins
 ---
-We're working on building a way for you to extend Praxis. Please join the
-[development mailing
-list](https://groups.google.com/forum/#!forum/praxis-development) if you'd like
-to join that conversation.
+
+## Overview
+
+Plugins provide a means to cleanly add new functionality to Praxis, either by extending the core classes with additional features, or by registering callbacks to run during request handling.
+
+For example, extensions to the core classes might include:
+
+ * Adding a `requires_authentication` directive to `ActionDefinition` to specify specify that the action requires an authenticated user. 
+ * Encapsulate access to the information about the currently logged-in user with a `Request#current_user` method that might return a `User` object.
+ * Inject a `before :action` filter into every `Controller` to perform authentication.
+
+
+### Anatomy of a Plugin
+
```

Comment:
(I would add this sentence before the "what follows...)

To highlight the pieces that plugins can choose to utilize, and the types of outcomes that they can achieve, it is best to start with a concrete example of a simple plugin.